### PR TITLE
Add documentation for C++ compiler

### LINKED ISSUE #10423: macOS: exceptions aren't caught when linking C++ to arm64 compiled code
I'm compiling OCaml bindings for a C++ library (https://github.com/grain-lang/binaryen.ml).   We previously hit a problem where the macOS compiled version wasn't capturing exceptions from the C++ library and was causing SIGABRT instead, as someone else demonstrates well here (https://github.com/vaivaswatha/OCamlCppExn) which was resolved by using keep_dwarf_unwind at the linking stage (issue #9026)

I'm now building an M1/arm64 native version, which compiles and builds fine but tests are failing because an exception thrown in the library is again aborting the program rather than being caught in the OCaml program.    Compiling and running https://github.com/vaivaswatha/OCamlCppExn but with keep_dwarf_unwind defined fails with the same problem as before:

libc++abi: terminating with uncaught exception of type std::runtime_error: This is not caught on macOS, why?

when it should hopefully have worked.

It may also be referenced here: https://git.brettgilio.com/ocaml/commit/482b7feb3743e140fd79c2b1bf929f2b738d42eb.html

Has anyone had success in getting arm64 compiled code to capture c++ exceptions?

This is on 4.12.0

### PR DESCRIPTION
**Pull request by phated**
I'm unsure if I did this correctly, but I wanted to open *something* since @xavierleroy suggested this be added to the docs after we closed #10423

Happy to make any changes you want on this.

### CODE REVIEW

File: manual/src/cmds/unified-options.etex [hunk 1]
```diff
@@ -97,7 +97,9 @@ compile modules separately.
 \item["-cc" \var{ccomp}]
 Use \var{ccomp} as the C linker \nat{called to build the final executable }
 \comp{when linking in ``custom runtime'' mode (see the "-custom" option)}
-and as the C compiler for compiling ".c" source files.
+and as the C compiler for compiling ".c" source files. When linking object
+files produced by a C++ compiler (such as "g++" or "clang++"), it is recommended
+to use "-cc g++" or "-cc clang++".
```
Comment (kit-ty-kate) [inline manual/src/cmds/unified-options.etex]: what about `-cc c++` to just get whichever standard C++ compiler is installed?

Comment (phated) [reply manual/src/cmds/unified-options.etex]: Like this?
```suggestion
to use "-cc c++".
```

Comment (kit-ty-kate) [reply manual/src/cmds/unified-options.etex]: yes, for example

Comment (phated) [reply manual/src/cmds/unified-options.etex]: Committed

**Review by kit-ty-kate — APPROVED** [review]

**Review by gasche — APPROVED** [review]
Approved on behalf of @kit-ty-kate. Thanks both for the documentation improvement.

**Pull request merged**